src/main/java/com/minecraftabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java

```diff
@@ -213,17 +213,18 @@ public BlockSubRegistryHelper(RegistryHelper parent, DeferredRegister<Item> item
 	}
 
 	/**
-	 * Creates and registers a {@link AbnormalsStandingSignBlock} and a {@link AbnormalsWallSignBlock} with an {@link AbnormalsSignItem}.
+	 * Creates and registers a {@link AbnormalsStandingSignBlock} and a {@link AbnormalsWallSignBlock} with a {@link SignItem}.
 	 *
 	 * @param name  - The name for the sign blocks
 	 * @param color - The {@link MaterialColor} for the sign blocks.
 	 * @return A {@link Pair} containing {@link RegistryObject}s of the {@link AbnormalsStandingSignBlock} and the {@link AbnormalsWallSignBlock}
 	 */
 	public Pair<RegistryObject<AbnormalsStandingSignBlock>, RegistryObject<AbnormalsWallSignBlock>> createSignBlock(String name, MaterialColor color) {
-		ResourceLocation texture = new ResourceLocation(this.parent.getModId(), "textures/entity/signs/" + name + ".png");
-		RegistryObject<AbnormalsStandingSignBlock> standing = this.deferredRegister.register(name + "_sign", () -> new AbnormalsStandingSignBlock(Block.Properties.create(Material.WOOD).doesNotBlockMovement().hardnessAndResistance(1.0F).sound(SoundType.WOOD), texture));
-		RegistryObject<AbnormalsWallSignBlock> wall = this.deferredRegister.register(name + "_wall_sign", () -> new AbnormalsWallSignBlock(Block.Properties.create(Material.WOOD, color).doesNotBlockMovement().hardnessAndResistance(1.0F).sound(SoundType.WOOD).lootFrom(standing.get()), texture));
-		this.itemRegister.register(name + "_sign", () -> new AbnormalsSignItem(standing.get(), wall.get(), new Item.Properties().maxStackSize(16).group(ItemGroup.DECORATIONS)));
+		WoodType type = WoodType.register(WoodType.create(this.parent.getModId() + ":" + name));
+		AbnormalsCore.WOOD_TYPES.add(type);
```

Comment:
This is not safe during parallel mod loading.
The `HashSet` could become corrupted easily. This is more reason to add a `SignManager` class with threadsafe methods.